README.md

```diff
@@ -329,33 +329,110 @@ OR
         pistachio # setenv bootcmd 'run nandboot'
         pistachio # saveenv
 
+##System upgrade
 
-####Configure Network
+Sysupgrade can now be used to flash ubifs images from within OpenWrt:
 
-You should see the logs on the console as below:
+    root@OpenWrt:/# sysupgrade -v /tmp/openwrt-pistachio-marduk-marduk_cc2520-ubifs.img
 
+You can download the ubifs image from webserver using wget or copy from USB drive.But the image must be put into /tmp as OpenWRT switches to a ramfs to do upgrade.
 
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  
-  BusyBox v1.24.1 (2016-03-04 17:43:09 IST) built-in shell (ash)
+###Downloading ubifs image from webserver
+    root@OpenWrt:/# cd /tmp
+    root@OpenWrt:/# wget http://192.168.91.79:8080/openwrt-pistachio-marduk-marduk_cc2520-ubifs.img
+(Replace `192.168.91.79:/8080` with IP address:/Portno of your webserver)
+###Copying ubifs image from USB drive
 
-  _______                     ________        __
- |       |.-----.-----.-----.|  |  |  |.----.|  |_
- |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
- |_______||   __|_____|__|__||________||__|  |____|
-          |__| W I R E L E S S   F R E E D O M
- -----------------------------------------------------
- DESIGNATED DRIVER (Bleeding Edge, r48138)
- -----------------------------------------------------
-  * 2 oz. Orange Juice         Combine all juices in a
-  * 2 oz. Pineapple Juice      tall glass filled with
-  * 2 oz. Grapefruit Juice     ice, stir well.
-  * 2 oz. Cranberry Juice
+    root@OpenWrt:/# mount /dev/sda1 /mnt/cd /tmp
+    root@OpenWrt:/# cp /mnt/openwrt-pistachio-marduk-marduk_cc2520-ubifs.img /tmp
+
+The image will be flashed onto the mtd partition that is not in use (firmware0 or firmware1) then uboot is updated to boot from that partition. This requires the firmware0 and firmware1 implementation in uboot with a default environment. The default bootloader binary is based upon U-Boot 2015.07-rc2-299ac94, has this implementation.
+
+Pre-requisite :
+U-boot environment variables should be set as follows:
+
+    pistachio # setenv bootcmd 'run dualnandboot'
+    pistachio # setenv bootfile uImage
+    pistachio # setenv fdtfile pistachio_marduk.dtb
 
-$root@OpenWrt:/#
-$root@OpenWrt:/#
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can also set or print the u-boot environment variables from OpenWrt too:
 
+    root@OpenWrt:/# fw_printenv
+    root@OpenWrt:/# fw_setenv bootcmd 'run dualnandboot'
+    root@OpenWrt:/# fw_setenv bootfile uImage
+    root@OpenWrt:/# fw_setenv fdtfile pistachio_marduk.dtb
 
+Uboot variable bootcount is reset after successful boot.You can cross-check by reading:
```

Comment:
This is not helpful, instead we could explain how and when uboot decides to fallback to another partition